apis/course/README.md

```diff
@@ -0,0 +1,576 @@
+        "position": {
+            "latitude": latitude of vessel at time of destination being set,
+            "longitude": longitude of vessel at time of destination being set
+        }
+    }
+}
+```
+
+### 3.3 Clear / Cancel a destination or Activated Route.
+---
+__Use case:__ Provide _"stop navigating to destination"_ or _"deactivate as route"_ operation.
+
+__Action:__ `DELETE`
+
+__Path:__ `/navigation/course/destination` or `/navigation/course/activeRoute`
+
+__Request body:__
+
+```JSON
+{
+    "value": null,
+    "source": Source making the change.
+}
+```
+
+This will result in the following Signal K path values being set:
+```JSON
+{
+    "activeRoute": {
+        "href": null,
+        "startTime": null
+    },
+    "nextPoint": {
+        "href": null,
+        "type": null,
+        "position": null,
+        "arrivalCircle": unchanged
+    },
+    "previousPoint": {
+        "href": null,
+        "type": null,
+        "position": null
+    }
+}
+```
+
+### 3.4 Activate a Route to follow.
+---
+
+__Use case:__ Provide _"activate / follow a route"_ operation.
+
+__Action:__ `PUT`
+
+__Path:__ `/navigation/course/activeRoute`
+
+__Request body:__
+```JSON
+{
+    "value": {
+        "href": "/resources/routes/urn:mrn:signalk:uuid:0d95e282-3e1f-4521-8c30-8288addbdbab",
+        "pointIndex": 0,
+        "reverse": false,
+        "arrivalCircle": 500
+    },
+    "source": Source making the change.
+}
+```
+where:
+- `href`: The path to the target route in `/resources/routes/`.
+- `pointIndex` (optional): Zero based index of the point within the route to use as the initial destination (defaults to 0 if not supplied or if value is larger than index of last point in the route).
+- `reverse` (optional): If `true` performs operations on route points in reverse order (defaults to false).
+- `arrivalCircle` (optional): Radius of circle centered at destination indicating arrival.
+
+This will result in the following Signal K path values being set:
+```JSON
+{
+    "activeRoute": {
+        "href": supplied href,
+        "startTime": Time at which route was activated (as per Signal K schema)
+    },
+    "nextPoint": {
+        "href": null,
+        "type": null if value not supplied,
+        "position": {
+            "latitude": latitude of route point at supplied index,
+            "longitude": longitude of route point at supplied index
+        },
+        "arrivalCircle": unchanged if value not supplied
+    },
+    "previousPoint": {
+        "href": null,
+        "type": null,
+        "position": {
+            "latitude": latitude of: vessel if supplied pointIndex=0 or point at poiintIndex-1,
+            "longitude": longitude of: vessel if supplied pointIndex=0 or point at poiintIndex-1
+        }
+    }
+}
+```
+
+### 3.5 Select point in the active Route as destination.
+---
+__Use case:__ Provide _"go to point in route"_ operation.
+
+__Action:__ `PUT`
+
+__Path:__ `/navigation/course/activeRoute/setPoint`
```

Comment:
Hmm. Should this be activeRoute/nextPoint/pointIndex and the payload just value? None of the other paths are setX, just X. 

And the payload could support absolute and relative operations `"value": 3` or`"changeWaypoint": "next"` or previous. 

The way increment/decrement are now with 1 and -1 can be confusing: if the route is reversed is "switch to next point in this reversed route" +1 or -1?

And add that property to the data model, so that it can be queried and is available.

Reply:
> Hmm. Should this be activeRoute/nextPoint/pointIndex and the payload just value? None of the other paths are setX, just X.

I have no problem with changing `setPont` to `pointIndex` and the value body being `{value: 3}`

I will update the document to refelct that.

Reply:
> The way increment/decrement are now with 1 and -1 can be confusing: if the route is reversed is "switch to next point in this reversed route" +1 or -1?

My thinking here is that indexes  used and returned via query are relative to the route direction and the api manages the adjustment whether route is being navigated in forward or reverse.

So for example  a route with 8 points activated in:
**Normal (forward) mode:**
- Index values reflect the order as defined within the route data 0= point at index 0.... 7= point at index 7.
- +1 next point of e.g. 0->1, 1->2,.....6->7
- -1 previous point e.g. 7->6, 6->5,.....1->0

**Reverse mode:**
- Start, previous and next operations reflect the sequence relative to the start of the journey... so in this case, 0= point at index 7, 1= point at index 6.... 7=point at index 0
- +1 go to next point e.g. 7->6, 6->5,.....1->0
- -1  go to previous point e.g. 7->6, 6->5,.....1->0

Doing it this way means the UI is not having to worry about number sequencing... start index is always 0 and end index is (length-1)

Reply:
Works for me. So the way `pointIndex` property works needs to be documented.

Should there be a way to retrieve `activeRoute`'s points as an array in one call, instead of fetching the href and reverse property and then fetching the points and accounting for the order in the client?

I just noticed that this PR uses both `current` and `active` route wording, we probably should use just `active`.